dns/route53/dns_test.go

```diff
@@ -2,79 +2,117 @@ package route53
 
 import (
 	"testing"
+	"time"
 
+	"fmt"
 	"net"
 
 	"github.com/aws/aws-sdk-go/aws"
-	route53Client "github.com/aws/aws-sdk-go/service/route53"
+	"github.com/aws/aws-sdk-go/service/route53"
 	log "github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestEditResourceRecords(t *testing.T) {
+const (
+	MOCKED_ZONE_ID = "20N31D"
+)
+
+type mockR53 struct {
+	recordSets map[string]*route53.ResourceRecordSet
```

Comment:
Care to elaborate on why this holds a map, and isn't just dumb enough to hold a single record set?

We only use this to mock tests, and I can't envision a unit test where we need to have to separate srv managers that use the same *instance* of this.

Also, if there's a good reason this needs to be singleton-ish, we should better index by zone id, not name, as names might be ends-with-dot sanitized or not.

Reply:
True, the manager only take care of one record. So, from that perspective it makes limited sense. Nevertheless, in my understanding, at the end `mockR53` represents a single zone, which - from a data model perspective - is a list of records, unique in `Name`. 

If that is plausible, I suggest to leave it, otherwise I'll remove it. Let me know.

Reply:
It just increases the complexity for no benefit. There's no code path ever acessing more than the (single) element in the hash map, and if we ever need to test multiple record managers, we can very well just instantiate multiple `mockR53`.